backends/kv/writer.go

```diff
@@ -548,31 +550,35 @@ func getUpdateExpressionFromRow(columns map[string]frames.Column,
 			return "", nil, nil, err
 		}
 
-		expression.WriteString(valueToTypedExpressionString(val, name))
+		expression.WriteString(valueWithNameToTypedExpressionString(val, name))
 	}
 
 	key := indexValFunc(index)
 	// Add key column as an attribute
-	expression.WriteString(valueToTypedExpressionString(key, indexName))
+	expression.WriteString(valueWithNameToTypedExpressionString(key, indexName))
 
 	var sortingVal interface{}
 	if sortingKeyName != "" {
 		sortingVal = sortingKeyValFunc(index)
-		expression.WriteString(valueToTypedExpressionString(sortingVal, sortingKeyName))
+		expression.WriteString(valueWithNameToTypedExpressionString(sortingVal, sortingKeyName))
 	}
 
 	return expression.String(), key, sortingVal, nil
 }
 
-func valueToTypedExpressionString(value interface{}, name string) string {
+func valueWithNameToTypedExpressionString(value interface{}, name string) string {
+	return fmt.Sprintf("%v=%v;", name, valueToTypedExpressionString(value))
```

Comment:
Since this is performance critical code afaict, consider writing directly to the string builder, avoiding the intermediate strings.